### PR TITLE
UI: Image fills aspect_ratio of texture dimensions

### LINKED ISSUE #13155: UI: image with Absolute position don't render correctly
## Bevy version

main since https://github.com/bevyengine/bevy/pull/10690

## What you did

Run example `game_manu` or `overflow_debug`

## What went wrong

Images that have `position_type: PositionType::Absolute` in their style are not displayed correctly

Screenshot 2024-05-01 at 02 23 48

### PR DESCRIPTION
**Pull request by bugsweeper**
# Objective

- Fixes #13155 .

## Solution

- Style::aspect_ratio is filled at same time as UiImageSize

## Testing

Run example  `game_menu`
image
 and `overflow_debug`
image

### CODE REVIEW

**Comment by bugsweeper** [comment]
@mockersf May I ask you for a review?

File: crates/bevy_ui/src/widget/image.rs [hunk 1]
```diff
@@ -107,6 +108,10 @@ pub fn update_image_content_size_system(
                     // multiply the image size by the scale factor to get the physical size
                     size: size.as_vec2() * combined_scale_factor,
                 }));
+                if let Some(mut style) = style {
```
Comment (alice-i-cecile) [inline crates/bevy_ui/src/widget/image.rs:111]: A comment here would make this code easier to follow.

File: crates/bevy_ui/src/widget/image.rs [hunk 2]
```diff
@@ -107,6 +108,10 @@ pub fn update_image_content_size_system(
...
+                    let Vec2 { x, y } = size.as_vec2();
+                    style.aspect_ratio = Some(x / y);
```
Comment (eidloi) [inline crates/bevy_ui/src/widget/image.rs:113]: shouldn't this be the other way around? Using the `style.aspect_ratio` controlling image sizing? I mean, I had the idea that the `Style` is always a _controlling_ component, never controlled by other components.

Comment (nicoburns) [reply crates/bevy_ui/src/widget/image.rs:113]: The way the web does it is that the intrinsic aspect ratio is separate to the aspect ratio style.

Comment (mockersf) [reply crates/bevy_ui/src/widget/image.rs:113]: same, not a fan of changing the `Style` component. Also, if the user set a specific aspect_ratio, we shouldn't overwrite it

Comment (bugsweeper) [reply crates/bevy_ui/src/widget/image.rs:113]: > The way the web does it is that the intrinsic aspect ratio is separate to the aspect ratio style.

I didn't find other place to set correct ratio, taffy knows nothing about type of nodes, the image should keep ratio, but div has no such restricktions. I agree that there shouldn't be overwriting, I will fix this.

Comment (nicoburns) [reply crates/bevy_ui/src/widget/image.rs]: > I didn't find other place to set correct ratio, if you know such way, please, feel free to describe it, taffy knows nothing about type of nodes

Taffy needs to change to fix this sensibly. Either Taffy needs to allow "measurable" items to communicate their inherent aspect ratio. Or Taffy needs to pass the styles in to the measure function to allow measurable items to apply it themselves.

Comment (bugsweeper) [reply crates/bevy_ui/src/widget/image.rs]: So we will not fix examples until Taffy become ready?

**Comment by alice-i-cecile** [comment]
Going with #13555.

**Pull request closed without merging**